cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -508,6 +512,12 @@ public ArtifactData apply(DatasetContext<Table> context) throws Exception {
             // column is the artifact name and version, value is the serialized PluginClass
             for (Map.Entry<byte[], byte[]> column : row.getColumns().entrySet()) {
               ArtifactColumn artifactColumn = ArtifactColumn.parse(column.getKey());
+              Id.Namespace artifactNamespace = artifactColumn.artifactId.getNamespace();
```

Comment:
Not sure if we had discussed this before, but there seems to be quite a bit of code duplication. Anyway to avoid it?